### PR TITLE
Vertical and horizontal airflow actions fix for Haier climate

### PR DESCRIPTION
**Pull request by paveldn**
# What does this implement/fix?

vertical_airflow and horizontal_airflow actions fix to fully support hOn protocol values.

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other

**Related issue or feature (if applicable):** fixes <link to issue>

**Pull request in [esphome-docs](https://github.com/esphome/esphome-docs) with documentation (if applicable):** esphome/esphome-docs#<esphome-docs PR number goes here>

## Test Environment

- [ ] ESP32
- [ ] ESP32 IDF
- [x] ESP8266
- [ ] RP2040

## Example entry for `config.yaml`:

```yaml
# Example config.yaml

```

## Checklist:
  - [x] The code change is tested and works locally.
  - [ ] Tests have been added to verify that the new code works (under `tests/` folder).

If user exposed functionality or configuration variables are added/changed:
  - [ ] Documentation added/updated in [esphome-docs](https://github.com/esphome/esphome-docs).

### CODE REVIEW

File: esphome/components/haier/hon_climate.cpp [hunk 1]
```diff
@@ -81,7 +81,7 @@ void HonClimate::set_outdoor_temperature_sensor(esphome::sensor::Sensor *sensor)
 AirflowVerticalDirection HonClimate::get_vertical_airflow() const { return this->vertical_direction_; };
 
 void HonClimate::set_vertical_airflow(AirflowVerticalDirection direction) {
-  if (direction > AirflowVerticalDirection::DOWN) {
+  if (direction > AirflowVerticalDirection::HEALTH_DOWN) {
     this->vertical_direction_ = AirflowVerticalDirection::CENTER;
   } else {
```
Comment (jesserockz) [inline esphome/components/haier/hon_climate.cpp]: I'm curious as to how its even possible to send a value higher than the highest possible value, especially given this is an enum class and compile errors will happen if you try to give a value that is not defined in the class...

**Review by jesserockz — CHANGES_REQUESTED** [review]

File: esphome/components/haier/hon_climate.cpp (hunk 1 above)
Comment (paveldn) [reply esphome/components/haier/hon_climate.cpp]: Just from the top of my head:
```
button:
  platform: template
  name: Test button
  on_press:
    then:
      lambda: |-
        haier::AirflowVerticalDirection d = haier::AirflowVerticalDirection::CENTER;
        d = (haier::AirflowVerticalDirection)(17);
        if (d > haier::AirflowVerticalDirection::HEALTH_DOWN)
            ESP_LOGD("test1", "OK");
        d = haier::AirflowVerticalDirection::CENTER;
        uint8_t t = 17;
        d = static_cast<haier::AirflowVerticalDirection>(t);
        if (d > haier::AirflowVerticalDirection::HEALTH_DOWN)
            ESP_LOGD("test2", "OK");
        d = haier::AirflowVerticalDirection::CENTER;
        d = *(haier::AirflowVerticalDirection*)&t;
        if (d > haier::AirflowVerticalDirection::HEALTH_DOWN)
            ESP_LOGD("test3", "OK");
        return;
```
tested and working in the latest 2023.7.0. Quite sure there are much more methods. 
But your house - your rules, I will remove checks.

Comment (jesserockz) [reply esphome/components/haier/hon_climate.cpp]: Who in their right mind is gonna do that? lol

Comment (paveldn) [reply esphome/components/haier/hon_climate.cpp]: I am 25 years in C++, I trust no one :)

**Review by jesserockz — APPROVED** [review]

**Review by jesserockz — APPROVED** [review]

**Pull request merged**